Ref.Windows.Tests/ViewModels/CopyReferenceDialogViewModelTests.cs

```diff
@@ -0,0 +1,137 @@
+            CitationStyle.Chicago, ReferenceOutputType.Markdown)]
+        public void VerifyCompleteArticle(string expected, CitationStyle style, ReferenceOutputType type)
+        {
+            var vm = new CopyReferenceDialogViewModel(new ArticleViewModel(TestUtility.CreateCounterexample()));
+            vm.CitationStyle = style;
+            vm.OutputType = type;
+
+            Assert.That(vm.Citation, Is.EqualTo(expected));
+        }
+
+        // APA
+        // Should actually be Silver, N.
+        [TestCase("Silver, Nate. (2012). The Signal and the Noise. Penguin Books.", CitationStyle.Apa, ReferenceOutputType.Plaintext)]
+        [TestCase("Silver, Nate. (2012). <i>The Signal and the Noise</i>. Penguin Books.", CitationStyle.Apa, ReferenceOutputType.Html)]
+        [TestCase("Silver, Nate. (2012). *The Signal and the Noise*. Penguin Books.", CitationStyle.Apa, ReferenceOutputType.Markdown)]
+        // Chicago
+        [TestCase("Silver, Nate. The Signal and the Noise. Penguin Books, 2012.", CitationStyle.Chicago, ReferenceOutputType.Plaintext)]
+        [TestCase("Silver, Nate. *The Signal and the Noise*. Penguin Books, 2012.", CitationStyle.Chicago, ReferenceOutputType.Markdown)]
+        public void VerifyMinimalBook(string expected, CitationStyle style, ReferenceOutputType type)
+        {
+            var book = new Book()
+            {
+                Author = "Silver, Nate",
+                Publisher = "Penguin Books",
+                Title = "The Signal and the Noise",
+                Year = "2012"
+            };
+            var vm = new CopyReferenceDialogViewModel(new BookViewModel(book));
+            vm.CitationStyle = style;
+            vm.OutputType = type;
+
+            Assert.That(vm.Citation, Is.EqualTo(expected));
+        }
+
+        // APA
+        [TestCase("Johnsonbaugh, R., Kalin, M. (1989). Applications Programming in C. New York, NY: Macmillan.",
+            CitationStyle.Apa, ReferenceOutputType.Plaintext)]
+        [TestCase("Johnsonbaugh, R., Kalin, M. (1989). <i>Applications Programming in C</i>. New York, NY: Macmillan.",
+            CitationStyle.Apa, ReferenceOutputType.Html)]
+        [TestCase("Johnsonbaugh, R., Kalin, M. (1989). *Applications Programming in C*. New York, NY: Macmillan.",
+            CitationStyle.Apa, ReferenceOutputType.Markdown)]
+        // Chicago
+        // Should actually be "Johnsonbaugh, R., and M. Kalin"
+        [TestCase("Johnsonbaugh, R., Kalin, M. Applications Programming in C. New York, NY: Macmillan, 1989.",
+            CitationStyle.Chicago, ReferenceOutputType.Plaintext)]
+        [TestCase("Johnsonbaugh, R., Kalin, M. *Applications Programming in C*. New York, NY: Macmillan, 1989.",
+            CitationStyle.Chicago, ReferenceOutputType.Markdown)]
+        public void VerifyCompleteBook(string expected, CitationStyle style, ReferenceOutputType type)
```

Comment:
There are several missing fields: edition, volume and translator. Should be added later.